### PR TITLE
Improved error handling and recovery in data reconcilers.

### PR DESCRIPTION
**Pull request by jortel**
This PR mainly improves lifecycle and error handling/recovery for provider (data) reconcilers.
While doing this, I noticed an opportunity to improve the controller `Reconciler` design pattern regarding event requeue.

requires: https://github.com/konveyor/controller/pull/23

### CODE REVIEW

File: pkg/controller/host/controller.go [hunk 1]
```diff
@@ -91,20 +101,23 @@ type Reconciler struct {
 //
 // Reconcile a Host CR.
 func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	fastReQ := reconcile.Result{RequeueAfter: FastReQ}
+	noReQ := reconcile.Result{}
 	var err error
 
 	// Reset the logger.
 	log.Reset()
+	log.SetValues("host", request.Name)
```
Comment (jortel) [inline pkg/controller/host/controller.go:110]: This adds the "provider": "_name_" to all subsequent log entries.

File: pkg/controller/provider/container/ocp/collection.go [hunk 2]
```diff
@@ -65,7 +70,6 @@ func (r *StorageClass) Reconcile() (err error) {
 //
 // Resource created watch event.
 func (r *StorageClass) Create(e event.CreateEvent) bool {
-	Log.Reset()
```
Comment (jortel) [inline pkg/controller/provider/container/ocp/collection.go:68]: The Reset() here in a different coroutine is no longer necessary and defeats the purpose of the correlation ID in the log entries like: "provider|<reconcile-id>" ...

File: pkg/controller/provider/container/ocp/collection.go [hunk 3]
```diff
@@ -128,7 +130,7 @@ func (r *NetworkAttachmentDefinition) Object() runtime.Object {
 //
 // Reconcile.
 // Achieve initial consistency.
-func (r *NetworkAttachmentDefinition) Reconcile() (err error) {
+func (r *NetworkAttachmentDefinition) Reconcile(ctx context.Context) (err error) {
```
Comment (jortel) [inline pkg/controller/provider/container/ocp/collection.go:133]: The context is passed in so if the reconciler is shutdown while iterating and inserting the initial list of resources, we won't try to insert into a database that may be closed/deleted by the shutdown().  The Shutdown() will _cancel_ the context.

File: pkg/controller/provider/container/vsphere/reconciler.go [hunk 4]
```diff
@@ -309,18 +310,24 @@ func (r *Reconciler) getUpdates(ctx context.Context) error {
 		This:    pc.Reference(),
 		Options: filter.Options,
 	}
+next:
 	for {
-		res, err := methods.WaitForUpdatesEx(ctx, r.client, &req)
+		response, err := methods.WaitForUpdatesEx(ctx, r.client, &req)
```
Comment (jortel) [inline pkg/controller/provider/container/vsphere/reconciler.go:315]: better readability.

File: pkg/controller/provider/container/vsphere/reconciler.go [hunk 5]
```diff
@@ -346,6 +353,10 @@ func (r *Reconciler) getUpdates(ctx context.Context) error {
 // Build the client.
 func (r *Reconciler) connect(ctx context.Context) error {
 	insecure := true
+	if r.client != nil {
```
Comment (jortel) [inline pkg/controller/provider/container/vsphere/reconciler.go:356]: Update connect() to be safely used for re-connect.

**Review by mansam — APPROVED** [review]

**Pull request merged**